src/ex-gen/Program.cs

```diff
@@ -133,91 +143,58 @@ private static void Run(string sourcePath, string inclusionFile, string exclusio
-
-                File.Delete(source);
-            }
-        }
+            string contents = $@"<Project Sdk=""Microsoft.NET.Sdk"">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>{tfm}</TargetFramework>
+  </PropertyGroup>
+  
+  <ItemGroup>
+    {
+    string.Join(Environment.NewLine, packageReferences.Select(pr => $@"<PackageReference Include=""{pr.id}"" Version=""{pr.version}"" />"))
+    }
+  </ItemGroup>
 
-        private static void ExtractArchives(string tempFolder)
-        {
-            var options = new ExtractionOptions { ExtractFullPath = true, Overwrite = true };
+</Project>";
 
-            foreach (var fileName in Directory.GetFiles(tempFolder))
-            {
-                Console.WriteLine("\t" + Path.GetFileName(fileName));
+            var emptyMain = @"internal static class Program
+{
+    public static void Main()
+    {
+    }
+}";
 
-                var extractedFolderName = Path.GetFileNameWithoutExtension(fileName).Replace(".tar", "");
-                var exractedFolderPath = Path.Combine(tempFolder, extractedFolderName);
+            var directoryPath = Path.GetDirectoryName(projectFilePath);
+            Directory.CreateDirectory(directoryPath);
 
-                using (var fileStream = File.OpenRead(fileName))
-                using (var archive = OpenArchive(fileStream))
-                {
-                    var selectedEntries = archive.Entries.Where(
-                        e => e.Key.StartsWith(@"./shared/Microsoft.NETCore.App/") ||
-                        e.Key.StartsWith(@"shared/Microsoft.NETCore.App/") ||
-                        e.Key.StartsWith(@"shared\Microsoft.NETCore.App\"));
-                    if (!selectedEntries.Any())
-                        throw new InvalidDataException($"No archive selected to be extracted from {fileName} at {tempFolder}");
-
-                    foreach (var entry in selectedEntries)
-                        entry.WriteToDirectory(exractedFolderPath, options);
-                }
+            File.WriteAllText(projectFilePath, contents);
 
-                File.Delete(fileName);
-            }
+            var programCs = Path.Combine(directoryPath, "Program.cs");
+            File.WriteAllText(programCs, emptyMain);
         }
 
-        private static IArchive OpenArchive(FileStream fileStream)
+        private static void PublishProject(string projectFilePath, string outputPath, string rid)
         {
-            var extension = Path.GetExtension(fileStream.Name);
-            switch (extension)
-            {
-                case ".tar":
-                    return TarArchive.Open(fileStream);
-                case ".zip":
-                    return ZipArchive.Open(fileStream);
-                default:
-                    throw new NotImplementedException($"Unknown file {extension}");
-            }
+            var command = "dotnet";
+            var args = $@"publish --output ""{outputPath}"" --runtime ""{rid}""";
```

Comment:
The current flow will never result in a rebuild. Basically, if the `dotnet-sdk` folder exists, it will just take its contents without running publish. And if the folder doesn't exist, it will generate the project and republish. IOW, the you trigger the publish by deleting the `dotnet-sdk` folder.

Reply:
Ok then.